### PR TITLE
Keep Athenz user domains lookup for 1min, per identity

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review. @bjorncs @tokle FYI. 

There is already some delay on Athenz' end, so adding another minute on our side shouldn't matter much? It would, however, make access control _much_ faster, for all requests using Athenz for authorisation, and also reduce the number of calls we do against Athenz by a great deal.

### CODE REVIEW

**Comment by jonmv** [comment]
Ah, I see there are other lookups we do, as well. Let me add more of this.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/filter/AthenzRoleFilter.java [hunk 1]
```diff
@@ -46,11 +50,13 @@
 
     private final AthenzFacade athenz;
     private final TenantController tenants;
+    private final ExecutorService executor;
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/filter/AthenzRoleFilter.java]: Should probably be shut down on deconstruction? (Not 100% on whether this applies to filters)

**Comment by jonmv** [comment]
I don't think it matters much. The controller JDisc isn't redeployed and reconfigured, like regular applications. Would have to make its superclass extend `AbstractComponent` to achieve. @bjorncs, any thoughts?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/filter/AthenzRoleFilter.java (hunk 1 above)
Comment (bjorncs) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/filter/AthenzRoleFilter.java]: That is correct. The filter should inherit `AbstractComponent` and do executor shutdown in `deconstruct()`.

**Review by mpolden — APPROVED** [review]

**Pull request merged**